domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/LayoutServiceImpl.java

```diff
@@ -59,4 +54,36 @@ public void createDisplayGroups(List<DisplayGroupEntity> displayGroups) {
         }
         displayGroupRepository.saveAll(displayGroups);
     }
+
+    private void validate(List<GenericLayoutEntity> genericLayouts) {
+        ValidationResult result = new ValidationResult();
+        for (GenericLayoutEntity genericLayoutEntity : genericLayouts) {
+            for (GenericLayoutValidator validator : genericLayoutValidators) {
+                result.merge(validator.validate(genericLayoutEntity));
+            }
+        }
+
+        for (GenericLayoutEntity genericLayoutEntity : genericLayouts) {
+            genericLayouts.stream()
+                .filter(e -> e != genericLayoutEntity)
```

Comment:
this code in terms of performance is O(n2). Any way we can improve the performance?

Reply:
There is a handful of GenericLayoutEntities and executed on definition import. I think this is not an issue.